File: GithubUserJava/app/src/main/java/com/example/myapplication/factory/MainViewModelFactory.java
package com.example.myapplication.factory;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

//import com.example.myapplication.data.local.datastore.SettingPreferences;
import com.example.myapplication.viewmodel.MainViewModel;

public class MainViewModelFactory extends ViewModelProvider.NewInstanceFactory {


    public MainViewModelFactory() {

    }

    @SuppressWarnings("unchecked")
    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        if (modelClass.isAssignableFrom(MainViewModel.class)) {
            return (T) new MainViewModel();
        }
        throw new IllegalArgumentException("Unknown ViewModel class: " + modelClass.getName());
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/factory/MainViewModelFactory.kt
package com.example.myapplication.factory

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.myapplication.data.local.datastore.SettingPreferences
import com.example.myapplication.viewmodel.MainViewModel

class MainViewModelFactory(private val pref: SettingPreferences) : ViewModelProvider.NewInstanceFactory() {

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(MainViewModel::class.java)) {
            return MainViewModel(pref) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class: " + modelClass.name)
    }
}